spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/HttpClientConfigurableHttpConnectionFactory.java

```diff
@@ -53,22 +67,81 @@ public HttpConnection create(URL url) throws IOException {
 
     @Override
     public HttpConnection create(URL url, Proxy proxy) throws IOException {
```

Comment:
@spencergibb `create()` here is called when jgit is about to connect to a repo. This is when we need to lookup the appropriately configured `HttpClientBuilder`. As you can see, all we have to go on is the URL. This URL param will have any placeholders filled in with actual values. So the trick here is looking up the correct `HttpClientBuilder` with nothing but this URL.